From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Mon, 9 May 2016 21:46:51 +0200
Subject: [PATCH] target Ubuntu: Blacklist rtc-efi if not supported.

Older ThunderX and possibly other UEFI-enabled targets do not support
rtc-efi properly, so they end up filling dmesg with useless
complaints about not being able to read system time.

Target OS (Ubuntu) deployment already has a snippet that blacklists
i2c_piix4 in certain scenarios, so provide a similar test for rtc_efi.
---
 cloud-init-templates/boothook_fuel_7.0_ubuntu.jinja2 | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/cloud-init-templates/boothook_fuel_7.0_ubuntu.jinja2 b/cloud-init-templates/boothook_fuel_7.0_ubuntu.jinja2
index 253be2d..9758376 100644
--- a/cloud-init-templates/boothook_fuel_7.0_ubuntu.jinja2
+++ b/cloud-init-templates/boothook_fuel_7.0_ubuntu.jinja2
@@ -49,8 +49,14 @@ cloud-init-per instance resolv_conf_head_nameserver /bin/sh -c 'echo nameserver
 # configure black module lists
 # virt-what should be installed
 if [ ! -f /etc/modprobe.d/blacklist-i2c_piix4.conf ]; then
-    ( (virt-what | fgrep -q "virtualbox") && echo "blacklist i2c_piix4" >> /etc/modprobe.d/blacklist-i2c_piix4.conf || :) && update-initramfs -u -k all
-    modprobe -r i2c_piix4
+    ( (virt-what | fgrep -q "virtualbox") && \
+        ( echo "blacklist i2c_piix4" >> /etc/modprobe.d/blacklist-i2c_piix4.conf && \
+        update-initramfs -u -k all && modprobe -r i2c_piix4 ) || :)
+fi
+if [ ! -f /etc/modprobe.d/blacklist-rtc_efi.conf ]; then
+    ( /bin/cat /sys/class/rtc/rtc0/time > /dev/null 2>&1 ) || \
+        ( echo "blacklist rtc_efi" >> /target/etc/modprobe.d/blacklist-rtc_efi.conf && \
+        update-initramfs -u -k all && modprobe -r rtc_efi )
 fi
 
 cloud-init-per instance conntrack_ipv4 /bin/sh -c 'echo nf_conntrack_ipv4 | tee -a /etc/modules'
